src/osip-monitoring-controller/src/main/java/edu/kit/pse/osip/monitoring/controller/MonitoringController.java

```diff
@@ -80,14 +121,442 @@ private void setupUI(Stage stage) {
         AboutDialog aboutDialog = new AboutDialog();
         currentView.setMenuAboutButtonHandler(event -> aboutDialog.show());
         currentSettingsView.setSettingsCancelButtonHandler(event -> currentSettingsView.hideSettingsWindow());
-//        currentSettingsView.setSettingsSaveButtonHandler(new SettingsSaveButtonHandler(currentSettings,
-//                currentView, currentSettingsView));
+        currentSettingsView.setSettingsSaveButtonHandler(event -> handleSaveButton(currentSettings,
+                currentView, currentSettingsView));
     }
 
     /**
      * Called when the last window is closed.
      */
     public void stop() {
-        throw new RuntimeException("Not implemented!");
+        System.out.println("Stopped monitoring thread");
+        for (TankContainer cont : tanks) {
+            try {
+                cont.client.disconnectClient();
+            } catch (UAClientException e) {
+                System.err.println(e.getMessage());
```

Comment:
Ja, ich weiß. Aber bei manchen Meldungen sagt die Message schon, dass ein Fehler beim Disconnecten auftrat. Dann würde da stehen "Error while disconnecting:  Unable to disconnect from servers." Aber ok, ich machs noch dazu.